### PR TITLE
Add simple interventions to the classifier

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://interventions.pfe-preview.zooniverse.org

First steps towards adding interventions/notifications to PFE projects.

- Add a redux store, and a few actions, for the Sugar client.
- Moves Sugar project channel subscriptions, and tests, over to Redux.
- Adds a simple notifications listener, which simply stores an array of received notifications.
- Adds a very simple Intervention component, which simply renders the latest Sugar message during the classification summary step. If summaries are hidden, it interrupts the 'Done' and 'Done & Talk' buttons in order to show a message.

TODO, either in this PR or later.

- [x] Display notifications to volunteers somewhere in the classifier.
- [x] Remove notifications from the store once they've been dismissed.
- [ ] Offer an option to unsubscribe from receiving notifications.
- [ ] Subscribe volunteers to channels by UserID, as well as by ProjectID.
- [ ] Style interventions and position them so that they won't be missed.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by chrislintott** [comment]
Note on unsubscribing from notifications - this would have to be very modular, so that people can unsubscribe from specific notifications associated with particular campaigns. Ideally this would be done on the notification ('Never show me notifications like this')

**Comment by eatyourgreens** [comment]
Previously, subscriptions were managed by experiments.zooniverse.org. Will there be an external service managing these campaigns, or do we manage subscriptions in Panoptes user preferences?
https://github.com/zooniverse/Panoptes-Front-End/blob/b1b0c36be22f66b61211a463a8a97093bddbaf24/app/lib/experiments-client.coffee#L229-L250

**Comment by eatyourgreens** [comment]
If you'd like to try this out, https://interventions.pfe-preview.zooniverse.org/ now loads with a single test message, which should show after your first classification on any project.

**Comment by eatyourgreens** [comment]
I broke the feedback tests. 😞

File: app/redux/ducks/interventions.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export function dismiss() {
+  return { type: DISMISS_NOTIFICATION };
+}
+
+export function injectSubjects() {
+  // Get new subjects and add them to the subject queue
+}
+
+sugarClient.on('notification', notify);
+sugarClient.on('subject-queue', injectSubjects);
```
Comment (eatyourgreens) [inline app/redux/ducks/interventions.js]: Action creators shouldn’t be invoked directly like this. The Sugar client should dispatch an action when it handles an event.

**Comment by srallen** [comment]
I might be able to take a look at this later in the week.

File: app/main.cjsx [hunk 2]
```diff
@@ -4,12 +4,17 @@ ReactDOM = require 'react-dom'
...
 style = require '../css/main.styl'
+{ sugarClient } = require 'panoptes-client/lib/sugar'
 
 # Redux
 `import { Provider } from 'react-redux';`
 `import configureStore from './redux/store';`
+`import { notify, injectSubjects } from './redux/ducks/interventions';`
 store = configureStore()
 
+sugarClient.on('notification', message => store.dispatch(notify(message)));
```
Comment (mcbouslog) [inline app/main.cjsx]: Getting ReferenceError on local and staged branch `message is not defined`?

Comment (eatyourgreens) [reply app/main.cjsx]: That'll be because this should be coffeescript, not ES6.

Comment (eatyourgreens) [reply app/main.cjsx]: I've added the missing brackets now.

**Comment by eatyourgreens** [comment]
`fakeMessage` was just for debugging. I’ll remove that commit before this gets merged.

**Comment by mcbouslog** [comment]
Sounds good. I'll assign to you then. It's ready for merge pending the removal of `fakeMessage`.

**Pull request merged**